File: 2020/day2/part_two.go
package main

import (
	"bufio"
	"log"
	"os"
	"strconv"
	"strings"
)

func main() {

	file, err := os.Open("input.txt")
	if err != nil {
		log.Fatal(err)
	}
	defer file.Close()

	cnt := 0

	scanner := bufio.NewScanner(file)
	for scanner.Scan() {

		line := scanner.Text()
		parts := strings.Split(line, " ")
		if len(parts) != 3 {
			log.Fatalf("error on line with content %s", line)
		}

		positions := strings.Split(parts[0], "-")
		positionA, err := strconv.Atoi(string(positions[0]))
		if err != nil {
			log.Fatal(err)
		}
		positionB, err := strconv.Atoi(string(positions[1]))
		if err != nil {
			log.Fatal(err)
		}

		policyChar := rune(parts[1][0])

		charCount := 0
		if positionA >= 1 && positionA <= len(parts[2]) {
			if rune(parts[2][positionA-1]) == policyChar {
				charCount++
			}
		}

		if positionB >= 1 && positionB <= len(parts[2]) {
			if rune(parts[2][positionB-1]) == policyChar {
				charCount++
			}
		}

		if charCount == 1 {
			cnt++
		}
	}

	log.Print(cnt)
}

File: 2021/day01/part_two.py
def main():

    depths = {}

    with open("input.txt") as f:
        i = 0

        for line in f:
            current_depth = int(line)

            depths[i] = current_depth
            if i - 1 >= 0:
                depths[i-1] += current_depth
            if i - 2 >= 0:
                depths[i-2] += current_depth

            i += 1

    cnt = 0
    for i in range(1, len(depths)):
        if depths[i] > depths[i-1]:
            cnt += 1

    print(cnt)


if __name__ == "__main__":
    main()
